src/nvim/ex_docmd.c

```diff
@@ -76,6 +76,16 @@
 #include "nvim/event/rstream.h"
 #include "nvim/event/wstream.h"
 
+/// The scope of a command.
+///
+/// The higher a number, the deeper a scope.
+enum {
+  SCOPE_EDITOR,     ///< Affects the entire instance of NeoVim.
+  SCOPE_TABPAGE,    ///< Affects on tab page.
+  SCOPE_WINDOW,     ///< Affects one window.
+  NUMBER_OF_SCOPES  ///< Total number of scoped defined.
```

Comment:
@HiPhish I know this, but still use `#define` in such cases because SCOPE_\* entries have the same kind. NUMBER_OF_SCOPES is different, it is not a valid scope, so it should not be in the enum.

Also

1. Why it is anonymous enum? I think that some variables may have enumeration type.
2. You must use `kScopeEditor` for constant names, **not** `SCOPE_EDITOR`. Check the [style guide](http://neovim.io/develop/style-guide.xml?showone=Enumerator_Names#Enumerator_Names), what you are using is the old naming inherited from Vim. Macros (like NUMBER_OF_SCOPES) on the other side still use ALL_CAPS naming.